tests/spread/extensions/glib/run/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Build and run a basic glib snap
+
+environment:
+  SNAP_DIR: ../snaps/glib-hello
+
+# This is a manual test until v2.36 of core is in stable
+manual: true
+
+prepare: |
+  sudo snap refresh --edge core
+
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+  sudo snap refresh --stable core
+
+execute: |
+  cd "$SNAP_DIR"
+  snapcraft
+  sudo snap install glib-hello_*.snap --dangerous
+  [ "$(glib-hello)" = "hello world" ]
+
+  # Verify that the extension command chain went through the proper setup procedure
+  snap_user_data="$HOME/snap/glib-hello/current"
+  [ -d "$snap_user_data/.config" ]
+  [ -d "$snap_user_data/.local" ]
+  [ -h "$snap_user_data/.themes" ]
+
```

Comment:
It would be useful to test the contents of:

 - $snap_user_data/.config/user-dirs.dirs
 - $snap_user_data/.config/fontconfig/fonts.conf
 - $snap_user_data/.config/ibus

Reply:
Alright I'll add that in. Can you think of a way to test that snap upgrades work as expected? What new content should be copied over?

Reply:
On upgrade there are a number of files under $SNAP_USER_DATA that are deleted and regenerated. I just verified that reinstalling the same snap from a local file results in a new revision (from x1 to x2), which will exercise the upgrade code path. So we could install the snap once, run it, then intentionally modify/delete some files under $SNAP_USER_DATA, then install it and run it again, and verify that the files have been regenerated.
We could also verify that the contents of `.last_revision` are updated after the second run.

Reply:
I agree, but at least as this test snap stands, I don't think it puts anything into the snap itself that gets copied over, is that true? What would I need to add to the snap in order to make sure that works as expected?

Reply:
No, that snap doesn't copy over any custom content to $SNAP_USER_DATA, however the desktop helpers still create some common files/symlinks that can be overwritten for testing purposes:
 - `.config/fontconfig/fonts.conf`
 - `.config/ibus/bus` (symlink)
 - `.config/dconf/user` (symlink)
 - `.last_revision`